### PR TITLE
fix: use correct hasOne on swift references

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

Include the `.hasOne` on `model` for swift when using references.

#### Issue #, if available

N/A

#### Description of how you validated changes

Unit test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/main/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts [hunk 1]
```diff
@@ -624,13 +624,19 @@ export class AppSyncSwiftVisitor<
...
+          const targetNameAttrStr = this.isCustomPKEnabled() && connectionInfo.targetNames
+            ? `targetNames: [${connectionInfo.targetNames.map(target => `"${target}"`).join(', ')}]`
+            : `targetName: "${connectionInfo.targetName}"`;
+          return `.hasOne(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
+            connectionInfo.connectedModel,
+          )}.keys.${this.getFieldName(connectionInfo.associatedWith)}, ${targetNameAttrStr})`;
+        } else {
+          return `.hasOne(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
+            connectionInfo.connectedModel,
+          )}.keys.${this.getFieldName(connectionInfo.associatedWith)})`;
```
Comment (palpatim) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts:638]: I haven't tracked down `getFieldName`, but will this work for an association with a CPK?

```graphql
type Primary @model {
  pk: ID! @primaryKey(sortKeyFields: ["sk1", "sk2"])
  sk1: ID!
  sk2: ID!
  relateds: [Related] @hasMany(references: ["primaryPk", "primarySk1", "primarySk2"])
  content: String
}
type Related @model {
  id: ID!
  content: String
  primaryPk: ID
  primarySk1: ID
  primarySk2: ID
  primary: Primary @belongsTo(references: ["primaryPk", "primarySk1", "primarySk2"])
}
```

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts [hunk 2]
```diff
@@ -2976,4 +2976,38 @@ describe('AppSyncSwiftVisitor', () => {
       expect(generatedMetaComment).toMatchSnapshot();
     });
   });
+  describe('custom references', () => {
+    it('sets the association to the references field for hasOne/belongsTo', () => {
+      const schema = /* GraphQL */ `
+        type Project @model {
+          projectId: ID! @primaryKey(sortKeyFields: ["name"])
+          name: String!
+          team: Team @hasOne(references: ["projectId"])
```
Comment (atierian) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:2985]:
```suggestion
          team: Team @hasOne(references: ["projectId", "projectName"])
```

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts [hunk 3]
```diff
@@ -2976,4 +2976,38 @@ describe('AppSyncSwiftVisitor', () => {
...
+        }
+        type Team @model {
+          teamId: ID! @primaryKey(sortKeyFields: ["name"])
+          name: String!
+          projectId: ID!
+          project: Project @belongsTo(references: ["projectId"])
```
Comment (atierian) [inline packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:2991]:
```suggestion
          projectId: ID!
          projectName: ID!
          project: Project @belongsTo(references: ["projectId", "projectName"])
```

File: packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts (hunk 2 above)
Comment (atierian) [reply packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts:2985]: The sort key defined on the Primary model `Project` needs to be explicitly defined the Related model `Team` as a property and included in references argument.

File: packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts (hunk 1 above)
Comment (dpilch) [reply packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts:638]: I've checked with the Swift team and they say that the `associatedWith` is not actually used by the swift client so having only one field should not affect the functionality.

**Review by sundersc — APPROVED** [review]

**Review by palpatim — APPROVED** [review]

File: packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts (hunk 1 above)
Comment (dpilch) [reply packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts:638]: After further discussion with the Swift team, the previous statements is incorrect. We will need to make an additional change to codegen.

**Pull request closed without merging**